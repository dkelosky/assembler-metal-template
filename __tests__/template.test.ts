process.env.ALLOW_CONFIG_MUTATIONS = "yes"; // value doesn't matter

import { execSync } from "child_process";
import * as fs from "fs";

const TIMEOUT = 20000;

let buildResponse: any;

describe("template tests", () => {

    beforeAll(() => {

        // build and run jcl to compile and bind all source
        const gen = execSync(`npm run genjcl`);
        console.log(gen.toString());
        let cmd = "zowe jobs submit lf \"./build/custom.jcl\" --directory \"./output\" --rfj"

        buildResponse = JSON.parse(execSync(cmd).toString());

    }, TIMEOUT)

    describe("build tests", () => {
        it("should build without errors", () => {
            expect(buildResponse.data.retcode).toBe("CC 0000");
        });
    });

    describe("execution tests", () => {
        it("should read all input lines", () => {
            const file = fs.readFileSync(`output/${buildResponse.data.jobid}/RUN/SYSPRINT.txt`).toString().trim();
            expect(file).toBe("@TEST");
        });

        it("show our control block is built correctly", () => {
            const responses = fs.readFileSync(`output/${buildResponse.data.jobid}/RUN/SNAP.txt`).toString().split('\n');
            let index;
            responses.forEach((response, x) => {
                if (response.indexOf("TEMPLATE SNAP") > -1) {
                    index = x + 1;
                }
            });

            expect(index).toBeDefined();

            if (index) expect(responses[index].substr(9, responses[index].length)).toMatchSnapshot();
        });
    });
});
