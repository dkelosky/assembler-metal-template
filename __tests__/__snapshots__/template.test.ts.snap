// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template tests execution tests show our control block is built correctly 1`] = `" 00000000 00000000 00000000 00000000                                          *................                *"`;
